server/routes/phase-allocation.js

```diff
@@ -22,25 +23,34 @@ router.get(
     keycloak.getUserInfoMiddleware(),
   ],
   asyncMiddleware(async (req, res) => {
-    const { hcapUserInfo: user } = req;
+    /**
+     * @type {import("../keycloak").hcapUserInfo} hcapUserInfo
+     */
+    const user = req.hcapUserInfo;
     const siteId = parseInt(req.params.id, 10);
 
-    // TODO: HCAP-1336 check that user has access to siteId
-    if (siteId) {
-      const result = await getAllSitePhases(siteId);
+    // Validate request
+    if (!siteId) return res.status(400).send('Invalid site ID');
+    const authorized =
```

Comment:
Would it be useful to pull this out to an external, helper function to be reused? Or is it not used enough?